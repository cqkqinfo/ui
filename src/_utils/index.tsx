export const sleep = (ms = 10) =>
  new Promise(resolve => setTimeout(resolve, ms));
