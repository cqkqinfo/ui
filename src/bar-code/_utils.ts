const bar = require('wxbarcode/demo/utils/barcode.js'); //引入生成条形码方法

function convert_length(length: number) {
  return Math.round((my.getSystemInfoSync().windowWidth * length) / 750);
}

export function barcode(
  id: string,
  code: string,
  width: number,
  height: number,
) {
  //封装展示条形码方法
  bar.code128(
    my.createCanvasContext(id),
    code,
    convert_length(width),
    convert_length(height),
  );
}
