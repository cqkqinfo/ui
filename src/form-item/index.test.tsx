import React, { useState } from 'react';
import FormItem from './index';
import Form from '../form';
import Radio from '../radio';
import Space from '../space';
import { sleep } from '../_utils';
import { fireEvent, render } from '@testing-library/react';
import { RuleType } from 'rc-field-form/es/interface';

const validator = async ({
  type,
  label,
  value,
}: {
  type: RuleType | 'phone' | 'idCard';
  label: string;
  value: string;
}) => {
  const App = () => {
    const [form] = Form.useForm();
    const [isFinish, setIsFinish] = useState(false);
    return (
      <Space>
        {isFinish ? 'finished' : ''}
        <Form form={form} cell onFinish={() => setIsFinish(true)}>
          <FormItem
            label={label}
            name={'idCard'}
            rules={[
              {
                type,
              },
              {
                required: true,
                message: `请输入${label}`,
              },
            ]}
          >
            <input placeholder={`请输入${label}`} />
          </FormItem>
        </Form>
        <div onClick={() => form.submit()}>提交</div>
      </Space>
    );
  };
  const { getByText, getByPlaceholderText } = render(<App />);
  const btn = getByText('提交');
  const input = getByPlaceholderText(`请输入${label}`);

  btn.click();
  await sleep();
  expect(getByText(`请输入${label}`));

  fireEvent.change(input, {
    target: { value: '23' },
  });
  btn.click();
  await sleep();
  expect(getByText(`请输入正确的${label}`));

  fireEvent.change(input, {
    target: { value },
  });
  btn.click();
  await sleep();
  expect(getByText('finished'));
};

describe('表单项', () => {
  it('渲染', () => {
    const { baseElement } = render(
      <Form cell onFinish={console.log}>
        <FormItem label={'号码类型'} name={'type'} initialValue={1}>
          <Radio.Group>
            <Radio value={1}>身份证</Radio>
            <Radio value={2}>手机号</Radio>
          </Radio.Group>
        </FormItem>
        <FormItem noStyle shouldUpdate>
          {(_, __, { getFieldValue }) => (
            <FormItem
              label={getFieldValue('type') === 1 ? '身份证号' : '手机号'}
              name={'number'}
              rules={[
                {
                  type: getFieldValue('type') === 1 ? 'idCard' : 'phone',
                  required: true,
                },
              ]}
            />
          )}
        </FormItem>
      </Form>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('身份证验证', () =>
    validator({
      type: 'idCard',
      value: '511602199808247696',
      label: '身份证号码',
    }));

  it('手机号验证', () =>
    validator({
      type: 'phone',
      value: '18580027047',
      label: '手机号',
    }));
});
