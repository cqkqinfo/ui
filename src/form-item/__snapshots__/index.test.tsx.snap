// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`表单项 渲染 1`] = `
<body>
  <div>
    <div
      class="space vertical"
      style="line-height: 1; padding: 0px 10px; align-self: stretch; box-sizing: border-box; background-color: rgb(255, 255, 255); border-radius: 10px; box-shadow: 0 0 10px rgba(39,128,217, 0.15);"
    >
      <div
        class="item cell"
        initialvalue="1"
      >
        <div
          class="labelWrap"
        >
          <div
            class="mark"
            style="opacity: 0;"
          >
            *
          </div>
          <div
            class="label"
            style="justify-content: flex-end;"
          >
            <div>
              号
            </div>
            <div>
              码
            </div>
            <div>
              类
            </div>
            <div>
              型
            </div>
          </div>
        </div>
        <div
          class="children"
          style="justify-content: flex-end;"
        >
          <div
            class="group"
          >
            <div
              class="container radio"
              style="background: none;"
            >
              <div
                class="dot"
                style="border-color: #eee; border-width: 1px; background: rgb(255, 255, 255);"
              />
              身份证
            </div>
            <div
              class="container radio"
              style="background: none;"
            >
              <div
                class="dot"
                style="border-color: #eee; border-width: 1px; background: rgb(255, 255, 255);"
              />
              手机号
            </div>
          </div>
        </div>
        <div
          class="after"
          style="display: none;"
        />
      </div>
      <div
        class="item cell"
      >
        <div
          class="labelWrap"
        >
          <div
            class="mark"
            style="opacity: 1;"
          >
            *
          </div>
          <div
            class="label"
            style="justify-content: flex-end;"
          >
            <div>
              手
            </div>
            <div>
              机
            </div>
            <div>
              号
            </div>
          </div>
        </div>
        <div
          class="children"
          style="justify-content: flex-end;"
        >
          <input
            class="web input"
            placeholder="请输入手机号"
            placeholderstyle="[object Object]"
            toutiao-selection-end="999"
            toutiao-selection-start="999"
            value=""
            wechat-confirm-type="done"
            wechat-cursor-spacing="0"
            wechat-placeholder-class="input-placeholder"
            wechat-selection-end="-1"
            wechat-selection-start="-1"
          />
        </div>
        <div
          class="after"
          style="display: none;"
        />
      </div>
    </div>
  </div>
</body>
`;
